src/modules/views/Nextcloud/NextcloudMoveView.jsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import {
+  Navigate,
+  useLocation,
+  useParams,
+  useNavigate,
+  useSearchParams
+} from 'react-router-dom'
+
+import { hasQueryBeenLoaded } from 'cozy-client'
+
+import { LoaderModal } from 'components/LoaderModal'
+import { getParentPath } from 'lib/path'
+import MoveModal from 'modules/move/MoveModal'
+import { useNextcloudFolder } from 'modules/nextcloud/hooks/useNextcloudFolder'
+import { useNextcloudInfos } from 'modules/nextcloud/hooks/useNextcloudInfos'
+import { useNextcloudPath } from 'modules/nextcloud/hooks/useNextcloudPath'
+
+const NextcloudMoveView = () => {
+  const { state, pathname } = useLocation()
+  const [searchParams] = useSearchParams()
+  const { sourceAccount } = useParams()
+  const path = useNextcloudPath()
+  const navigate = useNavigate()
+
+  const { rootFolderName } = useNextcloudInfos({ sourceAccount })
+  const { nextcloudResult } = useNextcloudFolder({
+    sourceAccount,
+    path
+  })
+  const { nextcloudResult: nextcloudParentResult } = useNextcloudFolder({
+    sourceAccount,
+    path: getParentPath(path)
+  })
+
+  const hasFileIds = state?.fileIds != undefined
+  if (!hasFileIds) {
+    const newPath = getParentPath(pathname) + `?${searchParams.toString()}`
+    return <Navigate to={newPath} replace />
+  }
+
+  if (
```

Comment:
Suggestion: Use the `getLoadedData` helper function to simplify the conditional rendering logic.
Explanation: The `getLoadedData` function checks if the query result has been loaded and returns the data if available. 
This helps in avoiding repetitive checks using `hasQueryBeenLoaded` and accessing `result.data`.
It also makes the conditional rendering logic cleaner and easier to understand.

```jsx
const entries = getLoadedData(nextcloudResult)?.filter(({ _id }) =>
  state.fileIds.includes(_id)
) || [];

if (getLoadedData(nextcloudResult) && (path === '/' || getLoadedData(nextcloudParentResult))) {
  // onClose handler to navigate to the parent path when the modal is closed.
  const onClose = () => {
    const newPath = `${getParentPath(pathname)}?${searchParams.toString()}`;
    navigate(newPath, { replace: true });
  };

  // Return the MoveModal component if the queries are loaded and the necessary data is available.
  return (
    <MoveModal
      showNextcloudFolder
      currentFolder={currentFolder}
      entries={entries}
      onClose={onClose}
    />
  );
}
```

Reply:
I took your suggestion and did something similar, but the function returns a boolean instead